src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/OffenderSearchServiceTest.kt

```diff
@@ -62,4 +71,93 @@ class OffenderSearchServiceTest {
     )
     assertThat(offenderWithoutAlerts.hasAcctOpen).isFalse
   }
+
+  private fun createWebClientMockResponses(vararg responses: List<OffenderSearchPrisoner>) =
+    WebClient.builder()
+      .exchangeFunction {
+        val page = UriComponentsBuilder.fromUri(it.url()).build().queryParams["page"]?.get(0)?.toIntOrNull()
+        val response = if (page != null && page < responses.size) {
+          ClientResponse.create(HttpStatus.OK)
+            .header("Content-Type", "application/json")
+            .body(
+              mapper.writeValueAsString(
+                mapOf(
+                  "content" to responses[page],
+                  "totalElements" to responses.sumOf { it.size },
+                  "last" to (page == responses.size - 1),
+                )
+              )
+            )
+            .build()
+        } else {
+          ClientResponse.create(HttpStatus.NOT_FOUND).build()
+        }
+        Mono.just(response)
+      }
+      .build()
```

Comment:
looks overly complicated but can’t find a simpler way to do it… ideas?

Reply:
As discussed - while constructing `WebClient.builder()` is a bit complicated it's potentially neater than attempting to use Mockito to do it, as we're trying to validate the following code in `OffenderSearchService`, I think your approach is ok
```
  page += 1
} while (!pageOfData.last)
```